### ISSUE TITLE
Support for rotation

### ISSUE THREAD

**Issue by lammertw** [vinnyoodles/DualSlideMenu#7]
Any chance you could add support for rotation while one of the side panels is open? Right now it doesn't reposition the main view.

**Comment by vincent144** [comment]
Yeah, that is definitely something that is needed. I'll start working on that

**Comment by vincent144** [comment]
@lammertw I've looked into this issue and there isn't a way for this pod to support rotation. Rotation support must be handled when initializing the left, center, and right view controllers through the use of constraints. I admit that my example project did not support rotation because I set the frame of the controllers to be the static size when in the portrait orientation. I believe the best solution is to override the view controller's `viewWillTransitionToSize` method and update the constraints accordingly.

Sorry for the delayed response as well.

**Comment by lammertw** [comment]
Thanks for the answer. I'll definitely try this.
I also made some changes to handle replacing the left and right view controllers. I'll probably make a pull request for that soon.

**Comment by vincent144** [comment]
Awesome

**Comment by lammertw** [comment]
I've reread your answer with a little bit more focus and I think you might misunderstand what I meant with supporting rotation. The only problem I have is that when you rotate while the left or right view is openen, the main view keeps the same offset. For example, the default offset is 150. If I open the left view in portrait the x position of the main view will be moved to 170 (320 - 150). If I then rotatie the device into landscape, the x position is still 170. Instead I believe it should recalculate 568 - 150 = 418 and then reposition the x from 170 to 418.

I've included some screenshots.
Opened menu in portait:
![screen shot 2016-06-12 at 20 52 01](https://cloud.githubusercontent.com/assets/826126/15993241/95ec9806-30df-11e6-93c9-f032b85a9ffb.png)

Rotate into landscape:
![screen shot 2016-06-12 at 20 52 05](https://cloud.githubusercontent.com/assets/826126/15993246/a72af888-30df-11e6-96e0-cb8436ca28f3.png)

Which should be the same as if the menu is opened from landscape in the first place:
![screen shot 2016-06-12 at 20 52 09](https://cloud.githubusercontent.com/assets/826126/15993247/b60543e0-30df-11e6-9874-99bc9b987cc4.png)

And worse is even when rotating from a menu opened in landscape to portrait because then the entire main view is gone:
![screen shot 2016-06-12 at 20 52 14](https://cloud.githubusercontent.com/assets/826126/15993250/d42130a0-30df-11e6-8d3c-8f098c055c53.png)

I don't think I can fix this by constraints in my views since the position of the main view is controlled by your library.

**Comment by vincent144** [comment]
Hmm, I'm starting to understand your issue now. So you're saying it should update the offset to accommodate the new orientation?

**Comment by lammertw** [comment]
Correct
